### ISSUE TITLE
MultivariateLinearRegression probably should not extend BaseRegression

### ISSUE THREAD

**Issue by Rainij** [mljs/regression-multivariate-linear#12]
It seems that BaseRegression was designed as a base class for univariate regression models. Therefore I think MultivariateLinearRegression should not extend BaseRegression.

I noticed this while using your package in a TypeScript file. The transpiler essentially complains about a missing method predict(x: number): number which is declared for BaseRegression but (of course) not for MultivariateLinearRegression. 

Could you please fix the issue or should I make a pull request? In any case, thanks for creating this package.

**Comment by targos** [comment]
I don't know, predict is overloaded in the definitions of this package: https://github.com/mljs/regression-multivariate-linear/blob/b959b170453e6fcd6979c3952cdd262b0e3973d7/regression-multivariate-linear.d.ts#L28
Did I do it wrong?

**Comment by Rainij** [comment]
Yes predict is overloaded but the compilation error comes from the fact that `predict(x: number): number` is "missing" in MultivariateLinearRegression (needed by BaseRegression).

To reproduce the error message one can take the code from the README, put it into a main.ts, compile it with tsc. tsc outputs the correct js file (not much to do) however it also gives the following error message: 

    error TS2416: Property 'predict' in type 'MultivariateLinearRegression' is not assignable to 
    the same property in base type 'BaseRegression'. 
    Type '{ (x: number[]): number[]; (x: number[][]): number[][]; (x: AbstractMatrix): Matrix; }' 
    is not assignable to type '{ (x: number): number; (x: number[]): number[]; }'.

    30     predict(x: AbstractMatrix): Matrix;
           ~~~~~~~
    Found 3 errors.

The error message is a little bit strange, but it would go away if you add `predict(x: number): number` near line 28. But of course, although this "fix" is convenient, that wouldn't make sense for multivariate regression. Therefore I thought that it might make sense to delete the inheritance relation to BaseRegression. I am not sure if it would make sense to introduce some "BaseMultivariateRegression". 

So to summarize I agree that the three overloads for predict are correct but the Inheritance is not - in my opinion.

**Comment by targos** [comment]
Thank you. Removing inheritance sounds good to me (both in the TS definitions and the JS code). I'd gladly accept a PR if you can do it.

**Comment by Rainij** [comment]
OK, I would be very glad to do it. Essentially I will only remove a few letters from the code (JS and TS as you said). Although its not very much to do I would like to do it tomorrow if its OK for you.
